embassy-boot/boot/src/lib.rs

```diff
@@ -679,7 +688,156 @@ impl FirmwareUpdater {
             self.dfu.from + offset + data.len()
         );
 
-        let mut write_offset = self.dfu.from + offset;
+        FirmwareWriter(self.dfu)
+            .write_block(offset, data, flash, block_size)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Prepare for an incoming DFU update by erasing the entire DFU area and
+    /// returning a `FirmwareWriter`.
+    ///
+    /// Using this instead of `write_firmware` allows for an optimized API in
+    /// exchange for added complexity.
+    pub fn prepare_update<F: NorFlash>(&mut self, flash: &mut F) -> Result<FirmwareWriter, F::Error> {
```

Comment:
From what I can tell, this is identical to prepare_update_blocking, so it should perhaps instead be async and take the async NorFlash trait instead?

Reply:
Huh, nice catch. A bit too fast on the copy-paste :)